x-pack/platform/plugins/shared/streams/server/routes/internal/streams/systems/route.ts

```diff
@@ -252,10 +258,88 @@ export const bulkSystemsRoute = createServerRoute({
   },
 });
 
+export const identifySystemsRoute = createServerRoute({
+  endpoint: 'POST /internal/streams/{name}/systems/_identify',
+  options: {
+    access: 'internal',
+    summary: 'Identify systems in a stream',
+    description: 'Identify systems in a stream with an LLM',
+  },
+  security: {
+    authz: {
+      requiredPrivileges: [STREAMS_API_PRIVILEGES.read],
+    },
+  },
+  params: z.object({
+    path: z.object({ name: z.string() }),
+    query: z.object({
+      connectorId: z.string(),
+      from: dateFromString,
+      to: dateFromString,
+    }),
+  }),
+  handler: async ({
+    params,
+    request,
+    getScopedClients,
+    server,
+    logger,
+  }): Promise<Observable<IdentifiedSystemsEvent>> => {
+    const {
+      systemClient,
+      scopedClusterClient,
+      licensing,
+      uiSettingsClient,
+      streamsClient,
+      inferenceClient,
+    } = await getScopedClients({
+      request,
+    });
+
+    await assertSignificantEventsAccess({ server, licensing, uiSettingsClient });
+
+    const {
+      path: { name },
+      query: { connectorId, from: start, to: end },
+    } = params;
+
+    const { write } = await checkAccess({ name, scopedClusterClient });
+
+    if (!write) {
+      throw new SecurityError(`Cannot update systems for stream ${name}, insufficient privileges`);
+    }
+
+    const [{ hits }, stream] = await Promise.all([
+      systemClient.getSystems(name),
+      streamsClient.getStream(name),
+    ]);
```

Comment:
so purpose is to provide prompt existing list of systems?

Reply:
yes, we will send it over as an existing system/partition